oauth2_http/javatests/com/google/auth/oauth2/GoogleCredentialsTest.java

```diff
@@ -192,6 +204,170 @@ public void fromStream_serviceAccountNoPrivateKeyId_throws() throws IOException
     testFromStreamException(serviceAccountStream, "private_key_id");
   }
 
+  @Test
+  public void fromStream_gdchServiceAccount_providesToken() throws IOException {
```

Comment:
This test failed since we could not pass the parameter of mock token server to create the GDCH credentials in fromStream method. This also applies on the test (getDefaultCredentials_envGdchServiceAccount_providesToken) in DefaultCredentialsProviderTest.

We do have test (fromJSON_hasAccessToken) for getting access token from a mock token server in GdchCredentialssTest. 

Could we skip the token exchange test here and in DefaultCredentialsProviderTest? Or we need to figure out how to pass the parameter.

Reply:
We don't need this test, it cannot test e2e as GoogleCredential. 

Instead lets add a test into GdchCredential that tests the token flow using GDCH credential directly. Basically, create the credential the same way ADC will do, provide a mock transport and test.

Reply:
done. thanks!

Reply:
Which one is that? I can't find a test that directly tests GDCH with mock transport and GetRequestMetadata

Reply:
It's the test `fromJSON_hasAccessToken()` in GdchCredentialsTest

Reply:
got it )